.github/workflows/main.yml

```diff
@@ -98,9 +99,23 @@ jobs:
         with:
           filters: .github/file-filters.yml
 
-  test_js:
+  test_js_client:
     if: needs.generate_clients.outputs.js_changed == 'true'
     name: JS Client
     needs: generate_clients
-    uses: ./.github/workflows/test-js.yml
+    uses: ./.github/workflows/test-js-client.yml
+    secrets: inherit
+
+  build_rust_client:
+    if: needs.generate_clients.outputs.rust_changed == 'true'
+    name: Rust Client
+    needs: generate_clients
+    uses: ./.github/workflows/build-rust-client.yml
+    secrets: inherit
```

Comment:
It is there since it builds the rust client in a "different" way. The test workflow uses `cargo test-sbf` while the build workflow uses the standard `cargo build`. It should not fail if the tests are fine, but `cargo build` is what will be used when the crate is compiled for release and it does not depend on the `sbf` toolchain.

Reply:
Got it, it makes sense to keep it then!